### PR TITLE
refactor: create wallet instance by config

### LINKED ISSUE #4183: create these wallet instances by a list of configs
> I guess in the future we could create these wallet instances by a list of configs, so that dont need to write this similar component.

Yes, as we have right now, at least 7 wallets in ecosystem, it would be really nice to have

ht @Jarsen136 

_Originally posted by @yangwao in https://github.com/kodadot/nft-gallery/issues/4178#issuecomment-1291701410_

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [x] Refactoring

### What's new?

- [x] PR closes #4183
- [ ] Requires deployment <rubick/magick_issue>
- [ ] <brief_description_of_what_I've_added>

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [ ] I've posted a screenshot of demonstrated change in this PR

### Optional

- [ ] I've tested it at </rmrk/collection/26902bc2f7c20c546a-1FVG7>
- [ ] I've tested PR on mobile and everything seems works
- [ ] I found edge cases
- [ ] I've written some unit tests 🧪

### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG)

### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot

- [ ] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

### CODE REVIEW

File: utils/config/wallets.ts [hunk 1]
```diff
@@ -1,24 +1,136 @@
...
+
 export enum SupportWalletExtension {
   PolkadotJs = 'polkadot-js',
   MetaMask = 'metamask',
   Clover = 'clover',
   Ledger = 'ledger',
-  Math = 'polkadot-js', // mathwallet
-  Nova = 'polkadot-js', // nova
+  Math = 'mathwallet',
+  Nova = 'nova',
```
Comment (roiLeo) [inline utils/config/wallets.ts:20]: iirc this should stay as `polkadot-js` to be launched on mobile app

**Review by roiLeo — COMMENTED** [review]
meh.

File: utils/config/wallets.ts (hunk 1 above)
Comment (Jarsen136) [reply utils/config/wallets.ts:20]: The logic was moved here
image

File: utils/config/wallets.ts [hunk 2]
```diff
@@ -1,24 +1,136 @@
...
+  source: SupportWalletExtension
+  walletUrl: string
+  guideUrl: string
+  isBrowserExtension: boolean
+  isMobileApp: boolean
+}
+type IWalletConfigMap = Partial<Record<SupportWalletExtension, WalletConfig>>
+
+export const WalletConfigMap: IWalletConfigMap = {
+  [SupportWalletExtension.PolkadotJs]: {
```
[the remark this answers is not in the document]
Comment (Jarsen136) [reply utils/config/wallets.ts:51]: will try to solve this

File: utils/config/wallets.ts (hunk 1 above)
Comment (Jarsen136) [reply utils/config/wallets.ts:20]: ### Enums should not have duplicate values
ref: https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1069

Comment (roiLeo) [reply utils/config/wallets.ts:20]: 522x0

**Comment by yangwao** [comment]
pay 50

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $50 @ 34.81 USD/KSM ~ 1.436 $KSM
🧗 [Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG](https://kusama.subscan.io/account/Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG)
🔗 [0xf9f846bedfb9766d4ac0d657110534e290ff8732f43ba6c85fb8bd381f19eaba](https://kusama.subscan.io/extrinsic/0xf9f846bedfb9766d4ac0d657110534e290ff8732f43ba6c85fb8bd381f19eaba)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**